geowebcache/core/src/main/java/org/geowebcache/layer/TileJSONProvider.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * <p>Copyright 2021
+ */
+package org.geowebcache.layer;
+
+import org.geowebcache.layer.meta.TileJSON;
+
+/** Interface to be implemented by layers which can provide a TileJSON document */
+public interface TileJSONProvider {
+
+    /* Returns true if TileJSON is supported or not compatible */
+    boolean supportsTileJSON();
```

Comment:
This method is useful only if it guarantees that getTileJSON returns a non null value, and it's at the same time faster than doing ``getTileJSON() != null``, otherwise I'd consider it redundant.

Reply:
In the MBTilesLayer implementation it always return true.
On a GeoServerTileLayer implementation I will return true if the underlying GridSet is 3857 or 900913.

Given a Layer the GetCapabilities will check if that layer supportsTileJSON and quickly skip it if not.

Would that make sense?

Reply:
Yessir!